OsmAnd/src/net/osmand/plus/track/TrackPointsCard.java

```diff
@@ -270,6 +278,7 @@ public void onPointsDeletionStarted() {
 	@Override
 	public void onPointsDeleted() {
 		updateGroups();
+		update();
```

Comment:
It calls adapter.synchronizeGroups() two times, firstly in updateGroups secondly in updateContent(). You can change updateGroups to clearSelectedGroups